### ISSUE TITLE
external syslog conf generated incorrectly rsyslog.conf.tt packetfence.conf

### ISSUE THREAD

**Issue by k1n6b0b** [inverse-inc/packetfence#3847]
### When adding a remote syslog server via the UI, a large number of duplicate entries (for general syslog facilities) are sent to the remote server even tho they are not selected.

The template for /etc/rsyslog.d/packetfence.conf is created as follows:
```
# mariadb_error.log
if $syslogtag contains "mysqld" then
  -/usr/local/pf/logs/mariadb_error.log
  @syslog.xxx.xxx:514
& stop
```
The proper syntax may be
```
# mariadb_error.log
if $syslogtag contains "mysqld" then
  -/usr/local/pf/logs/mariadb_error.log
  -@syslog.xxx.xxx:514
& stop
```
**Note:** the additional dash (-)

I am unable to find the appropriate rsyslog documentation discussing this particular IF format so my resolution may be incorrect however the behavior I am seeing currently is a duplicate message for things like AUTHPRIV.NOTICE, SYSLOG.ERR etc for each PF log I 'select' in the UI

I tried to edit the rsyslog.conf.tt file however it appears the `[% action %]` stored in the database for the filename contains the - while the remote syslog server entry does not.

Hopefully someone more familiar with the codebase and rsyslog can correct this (thank you kind soul!)

Packet capture showing behavior here: [https://imgur.com/a/Lr76ecJ](https://imgur.com/a/Lr76ecJ)

Also the FAQ documentation appears to be very outdated referencing log4perl: [https://packetfence.org/support/faq/packetfence-and-remote-syslog.html](https://packetfence.org/support/faq/packetfence-and-remote-syslog.html)

**Comment by nqb** [comment]
Hello @k1n6b0b,

I'm able to reproduce the issue.

@jrouzierinverse will confirm but for me, `rsyslog.conf.tt` is used to generate the initial rsyslog.d configuration in `/etc/rsyslog.d/packetfence.conf` where template substitution is done.

When configuring a Syslog Forwarder, PacketFence create a `/usr/local/pf/conf/syslog.conf`. I suppose PF will not reuse the `rsyslog.conf.tt`.

**Comment by julsemaan** [comment]
rsyslog.conf.tt is used to generate /etc/rsyslog/packetfence.conf using lib/pf/constants/syslog.pm as well and some other code.

@k1n6b0b don't worry, you're not the only one that struggles with the rsyslog documentation. I think they beat us in the lack of docs and assuming everyone is an expert when reading 2 lines that explains something hyper technical without context

Anyway, your findings are great and I believe it should be easily adjustable by @jrouzierinverse

**Comment by jrouzierinverse** [comment]
Fixed by eafea1e87baea567e55b803e129a38653107e38a

**Comment by julsemaan** [comment]
Adding the - just makes things asynchronous and this change broke forwarding completely and its not forwarding anymore with the latest maintenance.

We'll need to revert eafea1e from the maintenance and find the proper fix
